lessons/module-2/introduction-to-typescript.md

```diff
@@ -0,0 +1,320 @@
+---
+title: "Introduction to TypeScript"
+length: 120
+tags: typescript, javascript, type-checking, interfaces
+module: 2
+---
+
+## Learning Goals
+* Understand what TypeScript is and how it compares to JavaScript
+* Learn the benefits of using TypeScript in development
+* Get familiar with TypeScript's type system, including basic types, functions, and interfaces
+* Understand how TypeScript can help prevent common JavaScript errors
+
+## Vocab
+- `TypeScript`: A typed superset of JavaScript that compiles to plain JavaScript.
+- `Static Type Checking`: The process of verifying the type correctness of a program at compile-time.
+- `Transpiling`: The process of converting TypeScript code into JavaScript code.
+- `Type Annotation`: A way to explicitly declare the type of a variable or function parameter.
+- `Interface`: A TypeScript structure that defines the shape of an object, specifying its properties and their types.
+
+## Introduction to TypeScript
+TypeScript is a superset of JavaScript that introduces static type definitions. This means that TypeScript code must be transpiled into JavaScript before it can be executed by a browser or Node.js. TypeScript provides all the features of JavaScript but adds optional static typing, which can help catch errors early in the development process.
+
+### Why Use TypeScript?
+1. **Type Safety**: TypeScript's type system helps catch type-related errors at compile time, rather than at runtime.
+2. **Enhanced IDE Support**: With TypeScript, you get better code completion, navigation, and refactoring support.
+3. **Better Documentation**: Types serve as documentation, making it easier to understand and use code written by others.
+4. **Scalability**: TypeScript is particularly useful for large codebases, where it can help manage complexity.
+
+## Comparing JavaScript and TypeScript
+JavaScript is dynamically typed, meaning variable types are determined at runtime. TypeScript, on the other hand, allows for static type checking, where types are checked at compile-time.
+
+<section class="call-to-action">
+### Example of JavaScript vs TypeScript
+
+**JavaScript:**
+```js
+let greeting = "Hello, world!";
```

Comment:
I like this example!

It could tee off a good convo about how there is really NO good use case for wanting to deliberately change a variable's data type.